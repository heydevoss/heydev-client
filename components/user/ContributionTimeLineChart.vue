<template>
  <div>
    <div class="card">
        <chartjs-line class="chart"
          :backgroundcolor="bgColor"
          :beginzero="beginZero"
          :bind="true"
          :bordercolor="borderColor"
          :data="data[radio]"
          :datalabel="dataLabel"
          :labels="labels[radio]"
          :option="options"
        />
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      beginZero: true,
      bgColor: "#34F6F2",
      borderColor: "#34F6F2",
      data: {
        month: [1, 9, 0, 10, 28],
        quant: [4, 5, 10, 2, 18]
      },
      dataLabel: "Commits",
      labels: {
        month: ["Janeiro", "Fevereiro", "Mar√ßo", "Abril", "Maio"],
        quant: [1, 5, 10, 15, 20]
      },
      options: {
          legend: {
            position: 'right'
          }
      },
      radio: "month"
    };
  }
};
</script>

<style scoped>
  .chart{
    width: 500px;
    height: 500px;
  }

  @media screen and (max-width: 820px) {
    .chart{
      width: 300px;
      height: 300px;
    }
  }
</style>