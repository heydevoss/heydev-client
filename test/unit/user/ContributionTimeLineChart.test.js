import { shallowMount } from '@vue/test-utils'
import ContributionTimeLineChart from '@/components/user/ContributionTimeLineChart.vue'

describe('<ContributionTimeLineChart/>', () => {
  const wrapper = shallowMount(ContributionTimeLineChart, {
    stubs: {
      'chartjs-line': '<div class="chartjs-line-test"></div>'
    }
  })

  test('should be a Vue instance and render correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render the chartjs-line', () => {
    expect(wrapper.html().includes('<div class="chartjs-line-test"></div>'))
  })
  it('should contain label contain Commits', () => {
    expect(wrapper.vm.dataLabel).toBe('Commits')
  })
})
