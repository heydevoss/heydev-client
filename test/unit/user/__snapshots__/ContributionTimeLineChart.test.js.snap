// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContributionTimeLineChart/> should be a Vue instance and render correctly 1`] = `
<div>
  <div
    class="card"
  >
    <div
      backgroundcolor="#34F6F2"
      beginzero="true"
      bind="true"
      bordercolor="#34F6F2"
      class="chartjs-line-test chart"
      data="1,9,0,10,28"
      datalabel="Commits"
      labels="Janeiro,Fevereiro,MarÃ§o,Abril,Maio"
      option="[object Object]"
    />
  </div>
</div>
`;
